lambdaclass/mirra_backend
# [GH-636] Add purchase costs to items

## Motivation

Users need a way to purchase their items, we'll allow many ways to pay the same item.
Closes #636 

## Summary of changes

- [Add item costs to item templates config](https://github.com/lambdaclass/mirra_backend/pull/650/commits/fd3f0a0217afd5e3326a4003b15284535c2abf8e)
- [Add new ItemCost migration, schema and assocs](https://github.com/lambdaclass/mirra_backend/pull/650/commits/3f24dfaf65406175a47a2e82058be89c8b20dbd8)
- [Add ItemCost to seeds](https://github.com/lambdaclass/mirra_backend/pull/650/commits/ddf759e31e5f4bdc496ce0fb9879d3c9ab7a1f56)
- [Add currencies behavior for current buy item endpoint](https://github.com/lambdaclass/mirra_backend/pull/650/commits/296b27459c17089dbed754ba66ccf7e0a4475bbb)

## How to test it?

- Create user:
```
{:ok, google_user} = GameBackend.Users.find_or_create_google_user_by_email("some_email")
```
- Add money 🤑 
```
GameBackend.Users.Currencies.add_currency_by_name_and_game(google_user.user.id, "Gold", 1, 4444)
```
- Buy item
```
curl -X PUT -d item_name="muflus_gold" -d currency_name="Gold" http://localhost:4001/curse/users/google_user.user.id/items/buy
```
(see `apps/game_backend/priv/items_templates.json` for item names)
- Equip item
```
curl -X PUT -d character_name="muflus" -d item_name="muflus_gold" http://localhost:4001/curse/users/google_user.user.id/items/equip
```
(see `apps/game_backend/priv/characters_config.json` for character names)

## Checklist
